components-api/src/main/scala/pl/touk/nussknacker/engine/api/component/AdditionalComponentsUIConfigProvider.scala

```diff
@@ -0,0 +1,37 @@
+package pl.touk.nussknacker.engine.api.component
+
+import pl.touk.nussknacker.engine.api.component.AdditionalComponentsUIConfigProvider.SingleComponentConfigWithoutId
+
+trait AdditionalComponentsUIConfigProvider extends Serializable {
```

Comment:
Please add an changelog entry about new API and add the API to the list here: https://nussknacker.io/documentation/docs/developers_guide/Basics/#other-spis-for-nussknacker-customization-documentation-will-follow-soon
Please also add a short documentation above disc class describing the purpose of this interface